### PR TITLE
Remove extra entry in experience level table and cap experience at level 50 requirement

### PR DESCRIPTION
**Pull request by DakkJaniels**

### CODE REVIEW

**Comment by DakkJaniels** [comment]
Also fixes mana and health gain for level 50 issue.

**Comment by ikonomov** [comment]
Can we also have the gold color for both 50th lvl number and the experience number at 50th lvl like it is here https://github.com/diasurgical/devilutionX/pull/3928/files? Source/panels/charpanel.cpp L134 and L137

**Comment by DakkJaniels** [comment]
I would prefer to leave this fix as is, and the stuff with colors can be pulled in as part of that original PR.

File: Source/player.cpp [hunk 1]
```diff
@@ -2432,8 +2432,10 @@ void NextPlrLevel(Player &player)
 	} else {
 		player._pStatPts += 5;
 	}
-
-	player._pNextExper = ExpLvlsTbl[player._pLevel];
+	if (player._pLevel < MaxCharacterLevel)
+		player._pNextExper = ExpLvlsTbl[player._pLevel];
+	else
+		player._pNextExper = ExpLvlsTbl[MaxCharacterLevel - 1];
```
Comment (AJenbo) [inline Source/player.cpp:2438]: Maybe cleaner
```suggestion
	player._pNextExper = ExpLvlsTbl[std::min(player._pLevel, MaxCharacterLevel - 1)];
```

File: Source/player.cpp [hunk 2]
```diff
@@ -2602,7 +2592,10 @@ void InitPlayer(Player &player, bool firstTime)
 	SpellID s = PlayersData[static_cast<size_t>(player._pClass)].skill;
 	player._pAblSpells = GetSpellBitmask(s);
 
-	player._pNextExper = ExpLvlsTbl[player._pLevel];
+	int level = player._pLevel;
+	if (player._pLevel >= MaxCharacterLevel)
+		level = MaxCharacterLevel - 1;
```
Comment (AJenbo) [inline Source/player.cpp:2597]:
```suggestion
	int level = std::min(player._pLevel, MaxCharacterLevel - 1);
```

File: Source/player.cpp [hunk 3]
```diff
@@ -2468,19 +2470,14 @@ void NextPlrLevel(Player &player)
...
-	int expLvlsTblSize = sizeof(ExpLvlsTbl) / sizeof(ExpLvlsTbl[0]);
-	if (player._pExperience >= ExpLvlsTbl[expLvlsTblSize - 1]) {
+	if (player._pLevel > MaxCharacterLevel) {
+		player._pLevel = MaxCharacterLevel;
 		return;
 	}
-
-	if (player._pHitPoints <= 0) {
+	if (player._pLevel == MaxCharacterLevel)
 		return;
```
Comment (AJenbo) [inline Source/player.cpp:2480]: Might as well use `player._pLevel >= MaxCharacterLevel` in the first condition.

**Comment by AJenbo** [comment]
Looks good, but I think there are some potentials for removing some duplicate code.

**Comment by ikonomov** [comment]
Thank you AJenbo! <3

**Comment by ikonomov** [comment]
And of course Dakk!

**Pull request merged**